src/error_handling_helpers.h

```diff
@@ -36,6 +36,7 @@
 #endif
 #include <assert.h>
 #include <stdlib.h>
+#define __STDC_WANT_LIB_EXT1__ 1
 #include <string.h>
 
 #include <rcutils/error_handling.h>
```

Comment:
same here? there is even `#define __STDC_WANT_LIB_EXT1__ 1` in line 35. probably we need to move the `#include <rcutils/error_handling.h>` to above?

Reply:
I don't have a strong opinion on it. I let you decide. I have looked into [ROS coding guide](https://docs.ros.org/en/rolling/The-ROS2-Project/Contributing/Code-Style-Language-Versions.html#id3). It is based on the [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html). And in the Google C++ Style Guide, the header order is as follows:

dir2/foo2.h (a header corresponding to your cpp file)
A blank line
C system headers (more precisely: headers in angle brackets with the .h extension), e.g., <unistd.h>, <stdlib.h>.
A blank line
C++ standard library headers (without file extension), e.g., <algorithm>, <cstddef>.
A blank line
Other libraries' .h files.
A blank line
Your project's .h files.

It looks to me that in all project headers, the other project headers shall be included in the end.

Reply:
@clalancette I like your idea.